### ISSUE TITLE
Support pydantic >= 2 

### ISSUE THREAD

**Issue by glesperance** [RobertCraigie/prisma-client-py#791]
## Problem

I've tried using prisma-client-py with `fastapi = { version = "^0.100.0"}`, but it's incompatible because it requires Pydantic 2.

## Suggested Solution

Support for Pydantic v2 in prisma-client-py would be ideal, ensuring compatibility with the latest FastAPI server.

## Alternatives

As of now, I've had to forego Prisma and instead use `psycopg` for my database queries.
## Additional context

```
Because no versions of prisma match >0.9.1,<0.10.0
 and prisma (0.9.1) depends on pydantic (>=1.8.0,<2.0.0), prisma (>=0.9.1,<0.10.0) requires pydantic (>=1.8.0,<2.0.0).
Because pydantic-extra-types (2.0.0) depends on pydantic (>=2.0b3)
 and no versions of pydantic-extra-types match >2.0.0, pydantic-extra-types (>=2.0.0) requires pydantic (>=2.0b3).
Thus, prisma (>=0.9.1,<0.10.0) is incompatible with pydantic-extra-types (>=2.0.0).
And because fastapi (0.100.0) depends on pydantic-extra-types (>=2.0.0)
 and no versions of fastapi match >0.100.0,<0.101.0, prisma (>=0.9.1,<0.10.0) is incompatible with fastapi (>=0.100.0,<0.101.0).
```

**Comment by vivekashok1221** [comment]
`fastapi==0.100.0` [still supports pydantic v1](https://fastapi.tiangolo.com/release-notes/#pydantic-v1). However, it'd be nice if prisma-client-py supported pydantic v2. FastAPI will deprecated pydantic v1 in the future.

**Comment by Faolain** [comment]
Completely agree, we want to migrate to the prisma client but this is currently a dealbreaker as we use pydantic 2 in the rest of our application :/

**Comment by Oblynx** [comment]
Indeed, this is a big issue for us too. @RobertCraigie if you get a moment to think about this, how much work do you think it will take? Plz let us know also what priority this has for you.

**Comment by RunOrVeith** [comment]
We also need this, as langchain will [deprecate support for pydantic v1 on 25th of august](https://python.langchain.com/docs/guides/pydantic_compatibility#langchain-pydantic-migration-plan)

**Comment by RobertCraigie** [comment]
Hey all, sorry for the delay here. I unfortunately haven't been able to spend much time working on Prisma Python recently.

However this is the highest priority item for me, I'll try and get support out as soon as possible. Please note that this will take a lot of work as we use Pydantic extensively throughout the library & rely on features that have been changed fundamentally in v2.

The auto-migration tool is sadly not very helpful as I want to maintain support for Pydantic v1 & it doesn't help with that.

**Comment by Oblynx** [comment]
Understood, thanks for all the effort you're putting into this in the first place! You've made a very useful tool.

**Comment by RobertCraigie** [comment]
Support for v2 has been released in `v0.10.0`, please give it a try and let me know if you run into any issues!

**Comment by Oblynx** [comment]
Wow, this was fast! I'll see if it works out of the box for our tests.
